src/stories/containers/users/delete-account/delete-account.tsx

```diff
@@ -14,57 +14,83 @@ import { toast } from 'react-toastify';
-      const data = await clientRequest?.request(gqlQuery, filter);
-      if (data.userDelete) {
-        toast(
-          ({ closeToast }) => <Notification icon={<CheckMark />} borderColor="#B6EDE7" handleClose={closeToast} />,
-          {
-            position: toast.POSITION.BOTTOM_CENTER,
-            autoClose: 3000,
-            hideProgressBar: true,
-            closeButton: false,
-          }
-        );
-        setTimeout(() => {
-          router.push('/auth/manage');
-        }, 3000);
-      } else {
-        toast(
-          ({ closeToast }) => (
-            <Notification
-              icon={<Warning />}
-              borderColor="#FBE1D9"
-              handleClose={closeToast}
-              message="There was some problem deleting your account"
-            />
-          ),
-          {
-            position: toast.POSITION.BOTTOM_CENTER,
-            autoClose: 3000,
-            hideProgressBar: true,
-            closeButton: false,
-          }
-        );
+      const { query: gqlQueryLogin, input } = LOGIN_REQUEST(user?.username || '', value);
+      const response = await request(GRAPHQL_ENDPOINT, gqlQueryLogin, input);
+      if (response) {
+        const { query: gqlQuery, filter } = USERS_DELETE_FROM_ADMIN(id as string);
+        const data = await clientRequest?.request(gqlQuery, filter);
+        if (data.userDelete) {
+          toast(
+            ({ closeToast }) => <Notification icon={<CheckMark />} borderColor="#B6EDE7" handleClose={closeToast} />,
+            {
+              position: toast.POSITION.BOTTOM_CENTER,
+              autoClose: 3000,
+              hideProgressBar: true,
+              closeButton: false,
+            }
+          );
+          setTimeout(() => {
+            router.push('/auth/manage');
+          }, 3000);
+        } else {
+          toast(
+            ({ closeToast }) => (
+              <Notification
+                icon={<Warning />}
+                borderColor="#FBE1D9"
+                handleClose={closeToast}
+                message="There was some problem deleting your account"
+              />
+            ),
+            {
+              position: toast.POSITION.BOTTOM_CENTER,
+              autoClose: 3000,
+              hideProgressBar: true,
+              closeButton: false,
+            }
+          );
+        }
       }
     } catch (error) {
-      console.log('error', error);
+      console.log({ value });
+      toast(
+        ({ closeToast }) => (
+          <Notification
+            icon={<Warning />}
+            borderColor="#FBE1D9"
+            handleClose={closeToast}
+            message="There was some problem deleting your account"
+          />
+        ),
+        {
+          position: toast.POSITION.BOTTOM_CENTER,
+          autoClose: 3000,
+          hideProgressBar: true,
+          closeButton: false,
+        }
+      );
     }
-  }, [clientRequest, id, router]);
+  }, [clientRequest, id, router, user?.username, value]);
```

Comment:
We should move most of this logic to an mvvm hook to follow the same pattern across the app

Reply:
You are right, I will move all the logic